app/components/close_help_post.py

```diff
@@ -19,23 +24,60 @@
 )
 
 
-async def mention_entity(entity_id: int) -> str | None:
-    output = await entity_message(
-        # Forging a message to use the entity mention logic
-        cast("dc.Message", SimpleNamespace(content=f"#{entity_id}"))
-    )
-    return output.content or None
-
+@final
+@app_commands.guild_only()
+class Close(commands.GroupCog, group_name="close"):
+    def __init__(self, bot: GhosttyBot) -> None:
+        self.description = "Mark current post as resolved."
+        self.bot = bot
+
+    @override
+    def interaction_check(self, interaction: dc.Interaction, /) -> bool:
+        user = interaction.user
+        if is_dm(user):
+            return False
+        if not (
+            isinstance((post := interaction.channel), dc.Thread)
+            and post.parent_id == self.bot.config.help_channel_id
+        ):
+            # Can only close help post channel
```

Comment:
```suggestion
            # Can only close posts in #help
```

The grammar, or should I say lack thereof, makes the sentence extremely obtuse, and it was wrong too (help channel posts, not help post channel: you can't close channels).